docs/InstrumentationEvents.md

```diff
@@ -25,74 +25,29 @@ Instrumentation Event:
 }
 ```
 
-## <a name="list"> List of available events:
-
-### <a name="consumer"></a> Consumer
-
-* consumer.events.HEARTBEAT  
-  payload: {`groupId`, `memberId`, `groupGenerationId`}
-
-* consumer.events.COMMIT_OFFSETS  
-  payload: {`groupId`, `memberId`, `groupGenerationId`, `topics`}
-
-* consumer.events.GROUP_JOIN  
-  payload: {`groupId`, `memberId`, `leaderId`, `isLeader`, `memberAssignment`, `duration`}
-
-* consumer.events.FETCH_START
-  payload: {}
-
-* consumer.events.FETCH
-  payload: {`numberOfBatches`, `duration`}
-
-* consumer.events.START_BATCH_PROCESS  
-  payload: {`topic`, `partition`, `highWatermark`, `offsetLag`, `offsetLagLow`, `batchSize`, `firstOffset`, `lastOffset`}
-
-* consumer.events.END_BATCH_PROCESS  
-  payload: {`topic`, `partition`, `highWatermark`, `offsetLag`, `offsetLagLow`, `batchSize`, `firstOffset`, `lastOffset`, `duration`}
-
-* consumer.events.CONNECT
+Events which have `payload.duration` defined, it represent the time between the current event and the last event.
 
-* consumer.events.DISCONNECT
 
-* consumer.events.STOP
-
-* consumer.events.CRASH
-  payload: {`error`, `groupId`}
-
-* consumer.events.REQUEST
-  payload: {
-    `broker`,
-    `clientId`,
-    `correlationId`,
-    `size`,
-    `createdAt`,
-    `sentAt`,
-    `pendingDuration`,
-    `duration`,
-    `apiName`,
-    `apiKey`,
-    `apiVersion`
-  }
+## <a name="list"> List of available events:
 
-* consumer.events.REQUEST_TIMEOUT
-  payload: {
-    `broker`,
-    `clientId`,
-    `correlationId`,
-    `createdAt`,
-    `sentAt`,
-    `pendingDuration`,
-    `apiName`,
-    `apiKey`,
-    `apiVersion`
-  }
+### <a name="consumer"></a> Consumer
 
-* consumer.events.REQUEST_QUEUE_SIZE
-  payload: {
-    `broker`,
-    `clientId`,
-    `queueSize`
-  }
+| order | event               | payload                                                                                                                                                                                                       | description |
+|-------|---------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------|
+|1      | REQUEST             | {`broker`, `clientId`, `correlationId`, `size`, `createdAt`, `sentAt`, `pendingDuration`, `duration`, `apiName`, `apiKey`, `apiVersion`} |Whenever a request is made to the broker |
```

Comment:
I don't think this order column makes sense to have. Some of the events have relation to each other, like obviously `START_BATCH_PROCESS` happens before `END_BATCH_PROCESS`, but `GROUP_JOIN`, `REQUEST`, `CONNECT`, `FETCH`, and `COMMIT_OFFSETS` happen at different times and really have very little relationship to each other. By documenting an order here, it becomes a bit of a public contract that I don't want to have to uphold. The events can happen at any time, unless there's a direct relationship between them.